src/db/repository/postgres/WishCardsRepository.ts

```diff
@@ -0,0 +1,67 @@
+import { Kysely, UpdateObject } from 'kysely';
+
+import { DB, Wishcards } from '../../types/generated/database';
+
+export type WishcardsUpdateParams = Omit<
+	UpdateObject<DB, 'wishcards'>,
+	'id' | 'created_at' | 'updated_at' | 'agency_id' | 'child_id'
+>;
+export type WishcardsCreateParams = Omit<
+	Wishcards,
+	'id' | 'created_at' | 'updated_at' | 'deleted_at' | 'status'
+>;
+
+export default class WishcardsRepository {
+	constructor(private readonly database: Kysely<DB>) {}
+
+	create(createParams: WishcardsCreateParams) {
+		return this.database
+			.insertInto('wishcards')
+			.values(createParams)
+			.returningAll()
+			.executeTakeFirstOrThrow();
+	}
+
+	getAll() {
+		return this.database.selectFrom('wishcards').selectAll().execute();
+	}
+
+	getByWishItemName(itemName: string) {
+		return this.database
+			.selectFrom('wishcards')
+			.innerJoin('items', 'wishcards.item_id', 'items.id')
+			.selectAll('wishcards')
+			.where('items.name', 'ilike', itemName)
+			.executeTakeFirstOrThrow();
+	}
+
+	getById(id: string) {
+		return this.database
+			.selectFrom('wishcards')
+			.where('wishcards.id', '=', id)
+			.executeTakeFirstOrThrow();
+	}
+
+	update(id: string, updateParams: WishcardsUpdateParams) {
+		return this.database
+			.updateTable('wishcards')
+			.set(updateParams)
+			.where('id', '=', id)
+			.returningAll()
+			.executeTakeFirstOrThrow();
+	}
+
+	delete(id: string) {
+		return this.database.deleteFrom('wishcards').where('id', '=', id).executeTakeFirstOrThrow();
```

Comment:
```suggestion
		return this.database
			.deleteFrom('wishcards')
			.where('id', '=', id)
			.executeTakeFirstOrThrow();
```

Reply:
we want to delete them in case the agency created them by mistake or, like we had it before, some troll agency decides to create 50+ wishcards with only amazon gift cards as links

Reply:
@Enubia We still don't want to support hard deletes right out of the box, our relations aren't built to handle it without it throwing errors since we don't cascade deletes. We can add that functionality later on fairly easily though.

Reply:
makes sense, then we'll add it later on if needed